deep_qa/training/text_trainer.py

```diff
@@ -173,7 +176,10 @@ def load_dataset_from_files(self, files: List[str]):
         has background information could call this method, then do additional processing on the
         rest of the list, for instance).
         """
-        return TextDataset.read_from_file(files[0], self._instance_type())
+        dataset_type_key = self.dataset_params.pop_choice("type", list(concrete_datasets.keys()),
+                                                          default_to_first_choice=True)
+        dataset_type = concrete_datasets[dataset_type_key]
+        return dataset_type.read_from_file(files[0], self._instance_type())
```

Comment:
You need to pass the remaining `dataset_params` in here.  Also, you probably can't do `pop_choice` like this - subsequent dataset reads would not have the right dataset type.  You should have a test somewhere (probably in `TestTextTrainer`) that reading two datasets with some simple parameters gives the same type.  As it is, if you say you want an SNLI dataset, your training data will be an SNLI dataset, but your validation data will be a TextDataset.